### ISSUE TITLE
Error? in unet3d/losses.py

### ISSUE THREAD

**Issue by Littlehhh** [wolny/pytorch-3dunet#39]
## Bug? in Function https://github.com/wolny/pytorch-3dunet/blob/6e5a24b6438f8c631289c10638a17dea14d42051/unet3d/losses.py#L289
In this line you commit that
https://github.com/wolny/pytorch-3dunet/blob/6e5a24b6438f8c631289c10638a17dea14d42051/unet3d/losses.py#L304
But next line 
https://github.com/wolny/pytorch-3dunet/blob/6e5a24b6438f8c631289c10638a17dea14d42051/unet3d/losses.py#L305
 Should it be 
`src = input.unsqueeze(1) ` ?

**Comment by danarteaga** [comment]
While you are correct, it does not matter in the context in this script as long as the batch size is set to 1 as below. As a result, the length of the first 2 dimensions will always be 1.

From hdf5.py, lines 329-331
# when training with volumetric data use **batch_size of 1** due to GPU memory constraints
`return {
    'train': DataLoader(ConcatDataset(train_datasets), batch_size=1, shuffle=True, num_workers=num_workers),
    'val': DataLoader(ConcatDataset(val_datasets), batch_size=1, shuffle=True, num_workers=num_workers)
}`

**Comment by Littlehhh** [comment]
> While you are correct, it does not matter in the context of this script as long as the batch size is set to 1 as below. As a result, the length of the first 2 dimensions will always be 1.
> 
> From hdf5.py, lines 329-331
> 
> # when training with volumetric data use **batch_size of 1** due to GPU memory constraints
> `return { 'train': DataLoader(ConcatDataset(train_datasets), batch_size=1, shuffle=True, num_workers=num_workers), 'val': DataLoader(ConcatDataset(val_datasets), batch_size=1, shuffle=True, num_workers=num_workers) }`

Thanks for your answer.  But during the test phase,  It will not to be 1
